### ISSUE TITLE
Help wanted - problems sourcing nix files…

### ISSUE THREAD

**Issue by hedefalk** [fisherman/spin#9]
I'm using bax to source nix on a Mac. I have this:

```
if test -e '/Users/viktor/.nix-profile/etc/profile.d/nix.sh'
  bax source '/Users/viktor/.nix-profile/etc/profile.d/nix.sh'
end
```

That's a general setup-file for nix that looks like this:

```
if [ -n "$HOME" ] && [ -n "$USER" ]; then

    # Set up the per-user profile.
    # This part should be kept in sync with nixpkgs:nixos/modules/programs/shell.nix

    NIX_LINK=$HOME/.nix-profile


    # Append ~/.nix-defexpr/channels to $NIX_PATH so that <nixpkgs>
    # paths work when the user has fetched the Nixpkgs channel.
    export NIX_PATH=${NIX_PATH:+$NIX_PATH:}$HOME/.nix-defexpr/channels

    # Set up environment.
    # This part should be kept in sync with nixpkgs:nixos/modules/programs/environment.nix
    export NIX_PROFILES="/nix/var/nix/profiles/default $HOME/.nix-profile"

    # Set $NIX_SSL_CERT_FILE so that Nixpkgs applications like curl work.
    if [ -e /etc/ssl/certs/ca-certificates.crt ]; then # NixOS, Ubuntu, Debian, Gentoo, Arch
        export NIX_SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
    elif [ -e /etc/ssl/ca-bundle.pem ]; then # openSUSE Tumbleweed
        export NIX_SSL_CERT_FILE=/etc/ssl/ca-bundle.pem
    elif [ -e /etc/ssl/certs/ca-bundle.crt ]; then # Old NixOS
        export NIX_SSL_CERT_FILE=/etc/ssl/certs/ca-bundle.crt
    elif [ -e /etc/pki/tls/certs/ca-bundle.crt ]; then # Fedora, CentOS
        export NIX_SSL_CERT_FILE=/etc/pki/tls/certs/ca-bundle.crt
    elif [ -e "$NIX_LINK/etc/ssl/certs/ca-bundle.crt" ]; then # fall back to cacert in Nix profile
        export NIX_SSL_CERT_FILE="$NIX_LINK/etc/ssl/certs/ca-bundle.crt"
    elif [ -e "$NIX_LINK/etc/ca-bundle.crt" ]; then # old cacert in Nix profile
        export NIX_SSL_CERT_FILE="$NIX_LINK/etc/ca-bundle.crt"
    fi

    if [ -n "${MANPATH-}" ]; then
        export MANPATH="$NIX_LINK/share/man:$MANPATH"
    fi

    export PATH="$NIX_LINK/bin:$PATH"
    unset NIX_LINK
fi
```

What's failing is:

`export NIX_PATH=${NIX_PATH:+$NIX_PATH:}$HOME/.nix-defexpr/channels`

The other env is correctly exported but just NIX_PATH is not.

Current workaround is just to do:

```
if test -e '/Users/viktor/.nix-profile/etc/profile.d/nix.sh'
  set -x NIX_PATH $NIX_PATH $HOME/.nix-defexpr/channels
  bax source '/Users/viktor/.nix-profile/etc/profile.d/nix.sh'
end
```

Sorry to report as a bug here, because I think it's just nix preference for bash that's troubling here, but maybe you have some insight?

**Comment by jorgebucaran** [comment]
I apologize for taking so long to get back to you. How would I go about reproducing this? What do you expect would happen if I run `export NIX_PATH=${NIX_PATH:+$NIX_PATH:}$HOME/.nix-defexpr/channels` with this tool?

**Comment by jorgebucaran** [comment]
Someone was having an issue in Nix with Replay and we just (might have) fixed it (#14). 

I haven't heard from you in a while here, so I'm closing.
